pkgs/applications/misc/nut/default.nix

```diff
@@ -63,12 +64,25 @@ stdenv.mkDerivation rec {
 
   postInstall = ''
     substituteInPlace $out/lib/systemd/system-shutdown/nutshutdown \
+      --replace /bin/sed "${gnused}/bin/sed" \
       --replace /bin/sleep "${coreutils}/bin/sleep" \
       --replace /bin/systemctl "${systemd}/bin/systemctl"
 
     for file in system/{nut-monitor.service,nut-driver-enumerator.service,nut-server.service,nut-driver@.service} system-shutdown/nutshutdown; do
     substituteInPlace $out/lib/systemd/$file \
-      --replace "$out/etc/nut.conf" "/etc/nut.conf"
+      --replace "$out/etc/nut.conf" "/etc/nut/nut.conf"
+    done
+
+    substituteInPlace $out/lib/systemd/system/nut-driver-enumerator.path \
+      --replace "$out/etc/ups.conf" "/etc/nut/ups.conf"
+
+    # Inject a default value for NUT_CONFPATH into all binaries since the way we build the package
+    # results in the binaries being hardcoded to check $out/etc/ups.conf instead of
+    # /etc/nut/ups.conf (where the module places the file). We also cannot use
+    # `--sysconfdir=/etc/nut` since that results in the install phase trying to install directly
+    # into /etc/nut which predictably fails
```

Comment:
I'm interested in how big the delta is to just patch the binary. It looks like a change to the `confpath` function would be sufficient. Consider adding that to the patches...

Reply:
There's a ton of binaries (pretty much one for each driver) and I didn't want to go off the deep end to patch them all :sweat_smile: 

Actually revisiting the original motivation of adding this makes me question whether it should be added in the first place (or rather if we should wait to add it once we know better). Allow me to elaborate:

* I was doing some testing of my configuration and noticed that `upsmon -c fsd` would correctly trigger a shutdown of my machine, but it would _not_ lead to my UPS being restarted (using the `usbhid-ups` driver)
* There's a `nutshutdown` script which is supposed to be invoked by systemd right before shutdown, and this script basically triggers `upsdrvctl shutdown`
* Using `upsdrvctl -t shutdown` before this patch would give me this error: `Can't open /nix/store/8ii28z17ys92py1bkhmac2hfdrsnzffg-nut-2.8.0/etc/ups.conf: Can't open /nix/store/8ii28z17ys92py1bkhmac2hfdrsnzffg-nut-2.8.0/etc/ups.conf: No such file or directory`
* Using `NUT_CONFPATH=/etc/nut upsdrvctl -t shutdown` would correctly show a dry run invocation of `/nix/store/8ii28z17ys92py1bkhmac2hfdrsnzffg-nut-2.8.0/bin/usbhid-ups -a upswired -k`
* However, the shutdown script would _still_ fail to trigger the UPS to shutdown, so I assumed that the driver invocation might be losing the environment somehow (maybe by changing users?) so I thought all binaries would need to be wrapped
* Upon further testing, this didn't actually help much of anything and I'm not sure how to debug it further since the shutdown script runs so late there's no logs or even file system for me to debug with

I ended up working around this by changing my `SHUTDOWNCMD` to explicitly shutdown the `ups{d,drv,mon}` services (needed to relinquish control of the USB device) and then explicitly invoking `upsdrvctl shutdown` (my UPS has a 90 second delay which is enough time for the monitor to finish shutting down before power is cut).

With that in mind I think we have two options:

1. Bring back the commit which only patches the `nutshutdown` script to set a default value of `NUT_CONFPATH` since it isn't part of any explicit systemd service. Even if it doesn't help/fix whatever issue I was hitting, it certainly doesn't "hurt" to have
2. Don't patch anything (for now) until we better understand what the actual fix is.

Happy to go with either approach above!

Reply:
Decided to go with 1. above to keep things simple, but happy to revisit if anyone disagrees!